### PR TITLE
Rename alignment arguments to `align_to` and make them default to `UNIX_EPOCH`

### LINKED ISSUE #343: Unify arguments and defaults used for alignment
### What's needed?

The resampler, moving window and ring buffer, all have an option to align the window to some date, but all use different argument names, and the defaults also are different.

### Proposed solution

We should use the same name: `align_to` in all of them, and the same default `UNIX_EPOCH` too.

### Use cases

* Resampler
* Moving Window
* Ring Buffer

### Alternatives and workarounds

_No response_

### Additional context

We should probably also share the code, documentation and tests for alignment. Maybe a module `frequenz.sdk.tiemseries.alignment` can be created for this, but this is subject to how things look while implementing this feature.

### PR DESCRIPTION
**Pull request by leandro-lucarella-frequenz**
This is to make it consistent with the resampler.

Fixes #343.

### CODE REVIEW

**Comment by leandro-lucarella-frequenz** [comment]
(based on #335, only the last 2 commits are relevant)

**Review by mathias-baumann-frequenz — COMMENTED** [review]
LGTM, but not approving yet, waiting for the other PR

**Comment by leandro-lucarella-frequenz** [comment]
Rebased.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
     Note that the parameter `sampling_period` has been renamed to `input_sampling_period` to better distinguish it from the sampling period parameter in the `resampler_config`.
 
-  * Use the `UNIX_EPOCH` as the default alignment.
+  * Rename the constructor argument `window_alignment` to `align_to` and change the default to `UNIX_EPOCH`. This is to make it more consistent with the `ResamplerConfig`.
```
Comment (daniel-zullo-frequenz) [inline RELEASE_NOTES.md:55]: This commit should keep the line `Use the `UNIX_EPOCH` as the default alignment.` rather than deleting it

Comment (daniel-zullo-frequenz) [reply RELEASE_NOTES.md:55]: ah nevermind, the information is included in the new note

**Review by daniel-zullo-frequenz — DISMISSED** [review]

**Review by matthias-wende-frequenz — DISMISSED** [review]

**Comment by leandro-lucarella-frequenz** [comment]
For merging as it was pre-approved (I only rebased this on the current head branch)

**Pull request merged**